CS3219-AY2425S1/cs3219-ay2425s1-project-g11
# Implement matching service

### Description
This PR implements the matching service and features:
- Redis key value store for shared storage between replicated pods
- matching algorithm
  - topics queue for the first 15 sec
  - difficulty queue for the next 30 sec (give enough time for users to be matched by difficulty)
- RabbitMQ connection

### How to test
```
const user = {
    _id: "123",
    name: "John Doe",
    topic: "Node.js",
    difficulty: "Medium",
  };
```
- Run the services using `docker-compose up --build`
- Send a query with this format to the message queue `matching-service`
- Test for topic match
  - Send the next query within the next 15 sec
  - Check the queues of user1's id and user2's id to see if there are any messages
- Test for difficulty match
  - Send the next queue after 15 sec
  - Check the queues of user1's id and user2's id to see if there are any messages
 - Test for no match
   - Do not send a query
   - Check the queue of user1's id for a no match status

Closes #49
